lib/datadog/tracing/client_ip.rb

```diff
@@ -0,0 +1,191 @@
+# typed: true
+
+require 'ipaddr'
+
+require_relative '../core/configuration'
+require_relative 'metadata/ext'
+require_relative 'span'
+
+module Datadog
+  module Tracing
+    # Common functions for supporting the `http.client_ip` span attribute.
+    module ClientIp
+      DEFAULT_IP_HEADERS_NAMES = %w[
+        x-forwarded-for
+        x-real-ip
+        x-client-ip
+        x-forwarded
+        x-cluster-client-ip
+        forwarded-for
+        forwarded
+        via
+        true-client-ip
+      ].freeze
+
+      # A collection of headers.
+      class HeaderCollection
```

Comment:
I feel like this could go to the `Core` namespace.

Reply:
Fixed